test/functional/tests/stats/test_ioclass_stats.py

```diff
@@ -9,328 +9,666 @@
+    IoClassConfigStats,
 )
+
 from core.test_run import TestRun
 from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
-from test_tools.fs_tools import Filesystem
+from test_tools.fs_tools import create_directory, Filesystem
 from test_utils.filesystem.file import File
 from test_tools.os_tools import sync
 from test_tools.udev import Udev
 from connection.utils.output import CmdException
 from type_def.size import Size, Unit
 
-IoClass = ioclass_config.IoClass
-
-mountpoint = "/tmp/cas1-1"
-cache_id = 1
-
 
 @pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
 @pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex(
+    "cache_mode",
+    CacheMode.with_traits(CacheModeTrait.InsertWrite | CacheModeTrait.InsertRead),
+)
 @pytest.mark.parametrize("random_cls", [random.choice(list(CacheLineSize))])
-def test_ioclass_stats_basic(random_cls):
+def test_ioclass_stats_basic(cache_mode: CacheMode, random_cls: CacheLineSize):
     """
-        title: Basic test for retrieving IO class statistics.
-        description: |
-          Check if statistics are retrieved only for configured IO classes.
-        pass_criteria:
-          - Statistics are retrieved for configured IO classes.
-          - Error is displayed when retrieving statistics for non-configured IO class.
-          - Error is displayed when retrieving statistics for out of range IO class id.
+    title: Basic test for retrieving IO class statistics.
+    description: |
+        Check if statistics are retrieved only for configured IO classes.
+    pass_criteria:
+      - Statistics are retrieved for configured IO classes.
+      - Error is displayed when retrieving statistics for non-configured IO class.
+      - Error is displayed when retrieving statistics for out of range IO class id.
     """
-
     min_ioclass_id = 11
     max_ioclass_id = 21
+    mountpoint = "/tmp/open_cas_test_data/cas1-1"
+    file_size_base = Unit.Blocks4096.value
+    IoClass = ioclass_config.IoClass
+
+    with TestRun.step("Prepare cache and core devices"):
+        cache_device = TestRun.disks["cache"]
+        core_device = TestRun.disks["core"]
+
+        cache_device.create_partitions([Size(500, Unit.MebiByte)])
+        core_device.create_partitions([Size(2, Unit.GibiByte)])
+
+        cache_device = cache_device.partitions[0]
+        core_device = core_device.partitions[0]
+
+    with TestRun.step("Create filesystem on core device"):
+        core_device.create_filesystem(Filesystem.ext4)
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(
+            cache_dev=cache_device,
+            cache_mode=cache_mode,
+            cache_line_size=random_cls,
+            force=True,
+        )
+        cache_id = cache.cache_id
+
+    with TestRun.step("Set cleaning policy to NOP and seqcutoff to never on cache"):
```

Comment:
seqcutoff => sequential cutoff
You can 'set sequential cutoff **policy** to never' or 'disable sequential cutoff'. 
I'd go for 'Disable cleaning and sequential cutoff'.